types/http-proxy/index.d.ts

```diff
@@ -16,8 +17,25 @@ import * as events from "events";
 import * as url from "url";
 import * as stream from "stream";
 
+type ProxyTarget = ProxyTargetUrl | ProxyTargetDetailled;
+
 type ProxyTargetUrl = string | Partial<url.Url>;
 
+interface ProxyTargetDetailled {
```

Comment:
Could you please point me the documentation or code that uses this extra properties. The `Partial<url.Url>` already define some of this and I didn't found the details on the documentation.

Reply:
Yes it's partially covered by `Partial<url.Url>` but some of them are just slightly different, e.g: port is a string in url.URL and the lib allow number
I had put one link in the pr but it's indeed not enough:
- https://www.npmjs.com/package/http-proxy#http---https-using-a-pkcs12-client-certificate
- https://github.com/http-party/node-http-proxy/blob/a3fe02d651d05d02d0ced377c22ae8345a2435a4/lib/http-proxy/common.js#L37
- https://github.com/http-party/node-http-proxy/blob/a3fe02d651d05d02d0ced377c22ae8345a2435a4/test/lib-http-proxy-common-test.js#L322

Reply:
Thanks for pointing the code. Merging the PR.